### PR TITLE
Properly remove resource in case of overwrite by move operation

### PR DESCRIPTION
**Pull request by shabanovd**

### CODE REVIEW

File: src/org/exist/storage/NativeBroker.java [hunk 1]
```diff
@@ -2579,8 +2579,7 @@ public void moveResource(final Txn transaction, final DocumentImpl doc, final Co
                 }
                 */
 
-                trigger.beforeDeleteDocument(this, transaction, oldDoc);
-                trigger.afterDeleteDocument(this, transaction, oldDoc.getURI());
+                removeResource(transaction, oldDoc);
```
Comment (dizzzz) [inline src/org/exist/storage/NativeBroker.java]: did the trigger call disappear?

Comment (shabanovd) [inline src/org/exist/storage/NativeBroker.java]: yes, because `removeResource` have it

File: test/src/org/exist/storage/MoveTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author wolf
+ *
```
Comment (adamretter) [inline test/src/org/exist/storage/MoveTest.java]: Is this a test you have copied from somewhere? or is this a test you have written?

**Comment by adamretter** [comment]
@shabanovd I have taken a first pass and I can't see how this helps. Let me study it again this afternoon...

File: test/src/org/exist/storage/MoveTest.java (hunk 2 above)
Comment (shabanovd) [inline test/src/org/exist/storage/MoveTest.java]: Yes, I copy & modify

Comment (shabanovd) [inline test/src/org/exist/storage/MoveTest.java]: fixed

File: test/src/org/exist/storage/MoveTest.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            broker.saveCollection(tx, test1);
+            
+            test2 = broker.getOrCreateCollection(tx, TEST_COLLECTION_URI.append("test2"));
+            broker.saveCollection(tx, test2);
+
+            IndexInfo info = test1.validateXMLResource(tx, broker, doc1Name, XML1);
+            test1.store(tx, broker, info, XML1, false);
+
+            info = test2.validateXMLResource(tx, broker, doc2Name, XML2);
+            test1.store(tx, broker, info, XML2, false);
```
Comment (adamretter) [inline test/src/org/exist/storage/MoveTest.java]: This should be `test2.store` instead of `test1.store`

**Comment by adamretter** [comment]
Okay I have this one in hand... just tidying up and I will merge. It's a good fix @shabanovd

**Pull request closed without merging**